src/System.Net.Http/src/System/Net/Http/MessageProcessingHandler.cs

```diff
@@ -12,6 +13,22 @@ namespace System.Net.Http
 {
     public abstract class MessageProcessingHandler : DelegatingHandler
     {
+        // Private class used to capture the SendAsync state in
+        // a closure, while simultaneously avoiding a tuple allocation.
+        private class SendState : TaskCompletionSource<HttpResponseMessage>
+        {
+            public MessageProcessingHandler self;
```

Comment:
You're right, `self` is kind of confusing since it might refer to the handler or `SendState` itself. Will update.